src/ast/codegen_llvm.cpp

```diff
@@ -1229,30 +1230,60 @@ void CodegenLLVM::visit(Probe &probe)
         case ProbeType::uprobe:
         case ProbeType::uretprobe:
         {
-            auto symbol_stream = std::istringstream(bpftrace_.extract_func_symbols_from_path(attach_point->target));
-            matches = bpftrace_.find_wildcard_matches("", attach_point->func, symbol_stream);
-            break;
+          auto symbol_stream = std::istringstream(bpftrace_.extract_func_symbols_from_path(attach_point->target));
+          matches = bpftrace_.find_wildcard_matches("", attach_point->func, symbol_stream);
+          break;
         }
         case ProbeType::tracepoint:
           matches = bpftrace_.find_wildcard_matches(attach_point->target,
                                                     attach_point->func,
                                                     "/sys/kernel/debug/tracing/available_events");
           break;
+        case ProbeType::usdt:
+        {
+          auto usdt_symbol_stream = USDTHelper::probe_stream(bpftrace_.pid_, attach_point->target);
+          matches = bpftrace_.find_usdt_wildcard_matches(attach_point->ns, attach_point->func, usdt_symbol_stream);
+          break;
+        }
         default:
           std::cerr << "Wildcard matches aren't available on probe type '"
                     << attach_point->provider << "'" << std::endl;
           return;
       }
       tracepoint_struct_ = "";
-      for (auto &match : matches) {
+      for (auto &match_ : matches) {
```

Comment:
For USDT probes, to uniquely identify them we need to treat the provider namespace as part of the identifier.

For usdt probes, `match_` will look like "PROVIDER:FNAME", and so they need specially handling the cases where:

* We need the full unique identifier (full_func_id)
* We need just the FNAME (func_id)

This logic is repeated in bpftrace.cpp, to be able to access the probe index using the same identifier.